app/src/main/java/com/edulectronics/tinycircuit/Models/Scenarios/ImplementedScenarios/Scenario2.java

```diff
@@ -31,36 +31,37 @@ public int getPrompt() {
 
     @Override
     public boolean isCompleted(Circuit circuit) {
-        boolean hasLightbulb = false;
-        boolean hasSwitch = false;
+        boolean hasResistor = false;
         boolean isFullCircle = false;
+        boolean lampIsOn = false;
 
         for (Component component : circuit.getAllComponents()) {
             if(component != null) {
-                if (component.getClass() == Lightbulb.class) {
-                    hasLightbulb = true;
-                    continue;
-                }
-                if(component.getClass() == Switch.class) {
-                    hasSwitch = true;
+                if(component.getClass() == Resistor.class) {
```

Comment:
Is het niet de bedoeling dat er 1 resistor, 1 powersource en 1 lightbulb is ipv een willekeurige hoeveelheid? Zo ja kan je denk ik het beste in de circuit class een getCount(component) kunt maken en vanuit, in dit geval, de scenario2 class een methode aanroept in de circuitController.

Oftewel iets als

scenarioX.class:
if (circuitController.getComponentCount(component) == ?)

circuitController.class:
public int getComponentCount(component) {
 return circuit.getComponentCount(component);
}

circuit.class:
public int getComponentCount(component) {return x}